packages/macros/src/babel/evaluate-json.ts

```diff
@@ -388,13 +389,53 @@ export class Evaluator {
       return { confident: true, value: moduleExists(path, this.state), hasRuntimeImplementation: false };
     }
     if (callee.referencesImport('@embroider/macros', 'getConfig')) {
-      return { confident: true, value: getConfig(path, this.state, 'package'), hasRuntimeImplementation: true };
+      return { confident: true, value: getConfig(path, this.state, 'package'), hasRuntimeImplementation: false };
     }
     if (callee.referencesImport('@embroider/macros', 'getOwnConfig')) {
-      return { confident: true, value: getConfig(path, this.state, 'own'), hasRuntimeImplementation: true };
+      return { confident: true, value: getConfig(path, this.state, 'own'), hasRuntimeImplementation: false };
     }
     if (callee.referencesImport('@embroider/macros', 'getGlobalConfig')) {
-      return { confident: true, value: getConfig(path, this.state, 'getGlobalConfig'), hasRuntimeImplementation: true };
+      // Check for getGlobalConfig().fastboot.isRunning, which is the only pattern in use where config actually needs to have a runtime implementation.
+      // For compatibility reasons we will continue to support that. All other cases of macro configs are static now.
+      let maybeFastbootMemberExpression = path.parentPath;
+      if (
+        maybeFastbootMemberExpression.isMemberExpression() ||
+        maybeFastbootMemberExpression.isOptionalMemberExpression()
+      ) {
+        let maybeFastbootProperty = maybeFastbootMemberExpression.isMemberExpression()
+          ? maybeFastbootMemberExpression.get('property')
+          : maybeFastbootMemberExpression.isOptionalMemberExpression()
+          ? maybeFastbootMemberExpression.get('property')
+          : assertNever(maybeFastbootMemberExpression);
```

Comment:
This looks weird, but I found no better way to make TS happy. I had it first as simple as this:

![image](https://github.com/embroider-build/embroider/assets/1325249/40076a10-eacf-43de-9a34-57d05aa27029)

Those tertiary expressions narrow down the type before hitting `.get()` to make it return the correct type.


The type definition of `.get()` would not deal well with a union type (`NodePath<MemberExpression> | NodePath<OptionalMemeberExpression>`  it seems, falling back to the second loosely typed one:

![image](https://github.com/embroider-build/embroider/assets/1325249/1952a935-94c5-40f1-91d4-0d98b9272b1e)